src/test/java/com/codejoust/main/api/GameTests.java

```diff
@@ -242,101 +161,62 @@ public void startGameProblemNotEnoughFound() throws Exception {
 
         ApiError ERROR = ProblemError.NOT_ENOUGH_FOUND;
 
-        MvcResult result = this.mockMvc.perform(post(String.format(START_GAME, roomDto.getRoomId()))
-                .contentType(MediaType.APPLICATION_JSON_VALUE)
-                .content(UtilityTestMethods.convertObjectToJsonString(request)))
-                .andDo(print()).andExpect(status().is(ERROR.getStatus().value()))
-                .andReturn();
-
-        String jsonResponse = result.getResponse().getContentAsString();
-        ApiErrorResponse actual = UtilityTestMethods.toObject(jsonResponse, ApiErrorResponse.class);
-
+        ApiErrorResponse actual = MockHelper.postRequest(this.mockMvc, TestUrls.startGame(roomDto.getRoomId()), request, ApiErrorResponse.class, ERROR.getStatus());
         assertEquals(ERROR.getResponse(), actual);
     }
 
     @Test
     public void startGameWrongInitiator() throws Exception {
-        User host = new User();
-        host.setNickname(NICKNAME);
+        UserDto host = TestFields.userDto1();
 
         CreateRoomRequest createRequest = new CreateRoomRequest();
-        createRequest.setHost(UserMapper.toDto(host));
-
-        MvcResult result = this.mockMvc.perform(post(POST_ROOM)
-                .contentType(MediaType.APPLICATION_JSON_VALUE)
-                .content(UtilityTestMethods.convertObjectToJsonString(createRequest)))
-                .andDo(print()).andExpect(status().isCreated())
-                .andReturn();
+        createRequest.setHost(host);
 
-        String jsonResponse = result.getResponse().getContentAsString();
-        RoomDto roomDto = UtilityTestMethods.toObject(jsonResponse, RoomDto.class);
+        RoomDto roomDto = RoomTestMethods.setUpRoomWithOneUser(this.mockMvc, host);
 
-        UserDto user = new UserDto();
-        user.setNickname(NICKNAME_2);
+        UserDto user = TestFields.userDto2();
         StartGameRequest request = new StartGameRequest();
         request.setInitiator(user);
 
         ApiError ERROR = RoomError.INVALID_PERMISSIONS;
 
-        result = this.mockMvc.perform(post(String.format(START_GAME, roomDto.getRoomId()))
-                .contentType(MediaType.APPLICATION_JSON_VALUE)
-                .content(UtilityTestMethods.convertObjectToJsonString(request)))
-                .andDo(print()).andExpect(status().is(ERROR.getStatus().value()))
-                .andReturn();
-
-        jsonResponse = result.getResponse().getContentAsString();
-        ApiErrorResponse actual = UtilityTestMethods.toObject(jsonResponse, ApiErrorResponse.class);
-
+        ApiErrorResponse actual = MockHelper.postRequest(this.mockMvc, TestUrls.startGame(roomDto.getRoomId()), request, ApiErrorResponse.class, ERROR.getStatus());
         assertEquals(ERROR.getResponse(), actual);
     }
 
     @Test
     public void runCodeSuccess() throws Exception {
```

Comment:
> 1. Comments and sections within the test or codebase. Perhaps, if the test is more than like 15 lines of pure code (no spaces), it is too complicated and should be broken up in some way. This could just be different comments breaking up the test.

Good suggestion! I'll add this to the main PR description

> 2. I think we may want to add constructors to the request and DTO objects, so we don't have to do all the "setAttribute" methods.

Yeah definitely - I think this is where the builders can actually come into play, which I didn't add in this PR, but maybe in a later one

> 3. I think at one point you mentioned having minimal assertions, and I think we can definitely work on this. Maybe just one, or max two, assertions per test?

Yeah that's a good ideal to strive for - I'll add that to the main description as well